src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -101,7 +102,6 @@ public GoogleLoginState(
 
     this.isLoggedIn = false;
     this.email = "";
-    connected = true; // assume we're connected until checkCredentials is called
 
     listeners = Lists.newLinkedList();
```

Comment:
I don't think this list is thread-safe?  I know in Eclipse we hit cases where listener lists were modified as a result of firing an event.  I suppose we don't often add/remove listeners, or frequently fire events.  Could use a CopyOnWriteArrayList or ConcurrentLinkedDeque?